### PR TITLE
Pass through arguments to dotnet build3

### PR DESCRIPTION
**Pull request by livarcocc**
cc @eerhardt @piotrpMSFT @brthor @sokket

### CODE REVIEW

File: run-build.sh [hunk 1]
```diff
@@ -59,6 +73,11 @@ while [[ $# > 0 ]]; do
     shift
 done
 
+# $args array may have empty elements in it.
+# The easiest way to remove them is to cast to string and back to array.
+temp="${args[@]}"
+args=($temp)
```
Comment (brthor) [inline run-build.sh]: Have you tested this with quoted parameters?

For example something like `-e "Hello World"` Should resolve to two elements in the `args` array, and not 3.

Comment (livarcocc) [inline run-build.sh]: Yes. This breaks the case above. However, we haven't run into any issues regarding this yet. So, I would say, let's keep it.

File: run-build.sh [hunk 2]
```diff
@@ -59,6 +73,13 @@ while [[ $# > 0 ]]; do
     shift
 done
 
+# $args array may have empty elements in it.
+# The easiest way to remove them is to cast to string and back to array.
+# This will actually break quoted arguments, arguments like 
+# -test "hello world" will be broken into three arguments instead of two, as it should.
```
Comment (brthor) [inline run-build.sh]: 👍

**Comment by brthor** [comment]
:shipit:

**Pull request merged**